documentation/asciidoc/computers/configuration/device-tree.adoc

```diff
@@ -914,6 +914,38 @@ The root hub port number that the boot device is connected to - possibly via oth
 
 The Logical Unit Number for the mass-storage device.
 
+==== NVMEM nodes
+The firmware provides read-only, in-memory copies of ports of the bootloader EEPROM via the https://www.kernel.org/doc/html/latest/driver-api/nvmem.html[NVMEM] Subsystem.
+Using this interface avoids having to access the EEPROM from the OS via SPI which can be problematic because the EEPROM chip-select pin is muxed analog audio.
+
+Each region appears as an NVMEM device under `/sys/bus/nvmem/devices/` with a named alias under `/sys/firmware/devicetree/base/aliases`.
+
+Example shell script code for reading an nvmem mode from https://github.com/raspberrypi/rpi-eeprom/blob/master/rpi-eeprom-update[rpi-eeprom-update]
+
+----
+blconfig_alias="/sys/firmware/devicetree/base/aliases/blconfig"
+blconfig_nvmem_path=""
+
+if [ -f "${blconfig_alias}" ]; then
+   blconfig_ofnode_path="/sys/firmware/devicetree/base"$(strings "${blconfig_alias}")""
+   blconfig_ofnode_link=$(find -L /sys/bus/nvmem -samefile "${blconfig_ofnode_path}" 2>/dev/null)
+   if [ -e "${blconfig_ofnode_link}" ]; then
+      blconfig_nvmem_path=$(dirname "${blconfig_ofnode_link}")
+      fi
+   fi
+fi
+----
+
+`blconfig`
+
+The `blconfig` alias refers to an nvmem device that stores a copy of the EEPROM config file in plaintext format.
```

Comment:
Should 'nvmem' here be capitalised as 'NVMEM' for consistency with the earlier usages?

Reply:
I don't think you need `in plaintext format` given the EEPROM config file is already just text. (The term plaintext is also associated with crypto, to contrast with cyphertext so perhaps best to avoid that given the page also mentions crypto.

Reply:
plaimtext was removed